newrelic/resource_newrelic_synthetics_private_location_test.go

```diff
@@ -0,0 +1,125 @@
+//go:build integration
+// +build integration
+
+package newrelic
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/newrelic/newrelic-client-go/pkg/entities"
+)
+
+func TestAccNewRelicSyntheticsPrivateLocation_Basic(t *testing.T) {
+	resourceName := "newrelic_synthetics_private_location.bar"
+	rName := acctest.RandString(5)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckNewRelicSyntheticsPrivateLocationDestroy,
+		Steps: []resource.TestStep{
+			// Test: Create
+			{
+				Config: testAccNewRelicSyntheticsPrivateLocationConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNewRelicSyntheticsPrivateLocationExists(resourceName),
+				),
+			},
+			// Test: Update
+			{
+				Config: testAccNewRelicSyntheticsPrivateLocationConfigUpdated(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNewRelicSyntheticsPrivateLocationExists(resourceName),
+				),
+			},
+			// Test: Import
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"account_id", "description", "domain_id", "guid", "key", "location_id", "name", "verified_script_execution"},
+			},
+		},
+	})
+}
+
+func testAccCheckNewRelicSyntheticsPrivateLocationExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no synthetics private location is set")
+		}
+
+		client := testAccProvider.Meta().(*ProviderConfig).NewClient
+
+		queryString := fmt.Sprintf("domain = 'SYNTH' AND type = 'PRIVATE_LOCATION' AND name = '%s'", rs.Primary.ID)
+
+		found, err := client.Entities.GetEntitySearchByQuery(entities.EntitySearchOptions{}, queryString, []entities.EntitySearchSortCriteria{})
+		if err != nil {
+			return err
+		}
+
+		for _, e := range found.Results.Entities {
+			if !strings.EqualFold(string(e.GetGUID()), rs.Primary.ID) {
+				return fmt.Errorf("synthetics private location not found: %v - %v", rs.Primary.ID, found)
+			}
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckNewRelicSyntheticsPrivateLocationDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ProviderConfig).NewClient
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "newrelic_synthetics_private_location" {
+			continue
+		}
+
+		queryString := fmt.Sprintf("domain = 'SYNTH' AND type = 'PRIVATE_LOCATION' AND name = '%s'", r.Primary.ID)
+
+		time.Sleep(10 * time.Second)
```

Comment:
Personally I'd rather see a retry mechanism similar to https://github.com/newrelic/terraform-provider-newrelic/blob/main/newrelic/resource_newrelic_entity_tags_test.go#L81 instead of hard sleep.

What do you think @mbazhlekova and @sanderblue?